src/modules/logcollector/CMakeLists.txt

```diff
@@ -1,10 +1,15 @@
 cmake_minimum_required(VERSION 3.22)
 
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_SOURCE_DIR}/../../vcpkg/scripts/buildsystems/vcpkg.cmake")
+set(VCPKG_MANIFEST_DIR ${CMAKE_SOURCE_DIR}/../../)
+
 project(Logcollector)
 
 include(../../cmake/CommonSettings.cmake)
 set_common_settings()
 
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
In the previous version `fPic` was used in many places, not only when building Logcollector. Should we consider adding it to `set_common_settings()`?

If not I would consider using:

```
set_target_properties(Logcollector PROPERTIES POSITION_INDEPENDENT_CODE ON)
```

Using `set` will enable the option for all subdirectories added, right now there's none, but if this grow they will get `fpic` by default.

Reply:
The fix you propose does not work properly. Temporarily, we will leave it as it is, although we will discuss it internally.

Reply:
Thanks for trying it out, I'll have a look too when we switch to work on improvements.